### PR TITLE
qmlformat

### PR DESCRIPTION
**Pull request by m-kuhn**
Uses a gh workflow to run a qml linter.

To fix code (the famos pre-commit hook) there is another script that runs in a Docker since no one is expected to have Qt 5.15 installed already.

### CODE REVIEW

**Comment by m-kuhn** [comment]
@3nids comments on the implementation? General hesitations?

Once we are ready to roll, I'll update the formatting.

One particular case where we need to watch out and add curly braces `{}` for now

```
if ( a )
  // comment
  doSomething()
else
  // xyz
  doSomethingElse()
```

will be changed to

```
  // comment
  // xyz
if ( a )
  doSomething()
else
  doSomethingElse()
```

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5d563db7257f26e5fa25191c303854a97447ad9b-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5d563db7257f26e5fa25191c303854a97447ad9b-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5d563db7257f26e5fa25191c303854a97447ad9b-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5d563db7257f26e5fa25191c303854a97447ad9b-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-e3a99cd65407fdc9853ab4da774c84537aa26d9b-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-e3a99cd65407fdc9853ab4da774c84537aa26d9b-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-e3a99cd65407fdc9853ab4da774c84537aa26d9b-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-e3a99cd65407fdc9853ab4da774c84537aa26d9b-x86_64.apk)

File: .github/workflows/qmlformat.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+on:
+  push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: opengisch/qmlformat-lint-action
```
Comment (3nids) [inline .github/workflows/qmlformat.yml]: you still need to commit, right?

**Comment by 3nids** [comment]
Nice!
But wouldn't it be nicer to auto-fix the files directly in the pull request branch (since we are working without forks anyway)?

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-aec7556c02655cfeb614f283e49bf6c9852481c9-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-aec7556c02655cfeb614f283e49bf6c9852481c9-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-aec7556c02655cfeb614f283e49bf6c9852481c9-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-aec7556c02655cfeb614f283e49bf6c9852481c9-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-f34372545971fc38b8c53173f2dba3783fe1908c-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-f34372545971fc38b8c53173f2dba3783fe1908c-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-f34372545971fc38b8c53173f2dba3783fe1908c-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-f34372545971fc38b8c53173f2dba3783fe1908c-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5a6a7b0f277182bc89bfe681361a7954ea2322a6-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5a6a7b0f277182bc89bfe681361a7954ea2322a6-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5a6a7b0f277182bc89bfe681361a7954ea2322a6-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-912-5a6a7b0f277182bc89bfe681361a7954ea2322a6-x86_64.apk)

File: .github/workflows/qmlformat.yml (hunk 1 above)
Comment (m-kuhn) [reply .github/workflows/qmlformat.yml]: all done

**Comment by m-kuhn** [comment]
> Nice!
> But wouldn't it be nicer to auto-fix the files directly in the pull request branch (since we are working without forks anyway)?

I was thinking about a comment to trigger this

**Pull request closed without merging**